### PR TITLE
Addon-interactions: Mock window in browser environments

### PR DESCRIPTION
**Pull request by IanVS**
Issue: https://github.com/storybookjs/storybook/issues/17516

## What I did

Importing `fn` directly from `jest-mock` does not work in a browser, since jest expects to run in node and mocks out `global` instead of `window` (https://github.com/facebook/jest/blob/465518a2fa17ce45f7e3280dd0f54351273432c4/packages/jest-mock/src/index.ts#L1200-L1202).  This breaks in vite, which does not polyfill `global` like webpack does.

The change here was to use the `global` package, as is used elsewhere in storybook, to determine which global to mock (`window` in browsers, `global` in node).   

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

I've tested this by building, copying `dist` into my project, and verifying that the interactions addon works correctly, whereas it crashes if I just install the addon from npm.

### CODE REVIEW

File: addons/interactions/src/preset/argsEnhancers.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import { Args, addons } from '@storybook/addons';
+import { window as globalWindow } from 'global';
 import { FORCE_REMOUNT, STORY_RENDER_PHASE_CHANGED } from '@storybook/core-events';
 import { AnyFramework, ArgsEnhancer } from '@storybook/csf';
 import { instrument } from '@storybook/instrumenter';
-import { fn } from 'jest-mock';
+import { ModuleMocker } from 'jest-mock';
+
+const { fn } = new ModuleMocker(globalWindow);
```
Comment (yannbf) [inline addons/interactions/src/preset/argsEnhancers.ts:8]: Why not use `globalThis` here?

**Comment by IanVS** [comment]
I spent a few hours trying to get E2E tests running locally, and I have to give up now.  :(. 

I'm running `yarn local-registry --port 6000 --open`, but no matter what I try, I always get errors like 

```
@storybook/addon-actions@npm:6.5.0-alpha.40: The remote server failed to provide the requested resource
➤ YN0035: │   Response Code: 404 (Not Found)
➤ YN0035: │   Request Method: GET
➤ YN0035: │   Request URL: http://localhost:6000/@storybook%2faddon-actions
```

during the `🎨 Initializing Storybook with @storybook/cli` phase.  I'll try a few shots-in-the-dark, but I'm guessing the error here has to do with however addon-actions is hooked up to automatically mock actions...

**Comment by IanVS** [comment]
e2e test is now passing.  :success:

**Comment by IanVS** [comment]
Hah, looks like jest took care of this in their latest alpha too. https://github.com/facebook/jest/pull/12447

**Review by yannbf — APPROVED** [review]
LGTM! Great stuff @IanVS

File: addons/interactions/src/preset/argsEnhancers.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import { Args, addons } from '@storybook/addons';
-import { window as globalWindow } from 'global';
 import { FORCE_REMOUNT, STORY_RENDER_PHASE_CHANGED } from '@storybook/core-events';
 import { AnyFramework, ArgsEnhancer } from '@storybook/csf';
 import { instrument } from '@storybook/instrumenter';
 import { ModuleMocker } from 'jest-mock';
 
-const { fn } = new ModuleMocker(globalWindow);
+const JestMock = new ModuleMocker(global);
```
Comment (IanVS) [inline addons/interactions/src/preset/argsEnhancers.ts:7]: Ugh, I screwed this up, it needed to be `globalThis`.  I'll need to send another PR to fix this.

**Pull request merged**